pkg/api/kubegraph/edges.go

```diff
@@ -197,3 +204,37 @@ func AddAllRequestedServiceAccountEdges(g osgraph.Graph) {
 		}
 	}
 }
+
+func AddHPAScaleRefEdges(g osgraph.Graph) {
+	for _, node := range g.NodesByKind(kubegraph.HorizontalPodAutoscalerNodeKind) {
+		hpaNode := node.(*kubegraph.HorizontalPodAutoscalerNode)
+
+		hpa, ok := hpaNode.Object().(*extensions.HorizontalPodAutoscaler)
+		if !ok {
+			return
+		}
+
+		switch hpa.Spec.ScaleRef.Kind {
+		case kubegraph.ReplicationControllerNodeKind:
+			syntheticRC := &kapi.ReplicationController{
+				ObjectMeta: kapi.ObjectMeta{
+					Name:      hpa.Spec.ScaleRef.Name,
+					Namespace: hpa.Namespace,
+				},
+			}
+			rcNode := kubegraph.FindOrCreateSyntheticReplicationControllerNode(g, syntheticRC)
```

Comment:
Won't we need to do the type introspection to create the synthetic node? Are you arguing for that logic to be centralized or that it doesn't need to exist?